stl/inc/iterator

```diff
@@ -827,12 +827,16 @@ private:
 
     public:
         explicit _Proxy(iter_reference_t<_Iter>&& _Right) noexcept(
-            is_nothrow_move_constructible_v<iter_value_t<_Iter>>) // strengthened
-            : _Keep(_STD move(_Right)) {}
+            is_nothrow_constructible_v<iter_value_t<_Iter>, iter_reference_t<_Iter>>) // strengthened
+            : _Keep(_STD forward<iter_reference_t<_Iter>>(_Right)) {}
 
         _NODISCARD const iter_value_t<_Iter>* operator->() const noexcept /* strengthened */ {
             return _STD addressof(_Keep);
         }
+
+        _NODISCARD const iter_value_t<_Iter>& operator*() const noexcept /* strengthened */ {
+            return _Keep;
+        }

```

Comment:
I'd prefer to have distinct arrow and post-increment proxy types rather than to have both types support both operations.